folding/src/expressions.rs

```diff
@@ -602,6 +602,42 @@ impl<C: FoldingConfig> FoldingCompatibleExpr<C> {
             _ => panic!("unsupported"),
         }
     }
+
+    /// Maps variable (column index) in expression using the `mapper`
+    /// function. Can be used to modify (remap) the indexing of
+    /// columns after the expression is built.
+    pub fn map_variable(
```

Comment:
This is cherry-picked [here](https://github.com/o1-labs/proof-systems/pull/2337)